### PR TITLE
Add Event stitching support

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Add event stitching support by using `EventBuilder.chainToParentEvent()` API when creating new events.
Updates Mobile Core dependency to v2.2.0

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 1]
```diff
@@ -149,15 +149,16 @@ void handleRulesEngineResponse(@NonNull final Event event) {
...
 	}
 
 	/**
 	 * Helper to create and dispatch an experience event.
 	 * @param data map containing free-form data to send to Edge Network
 	 * @param timestamp timestamp of Event
+	 * @param event the parent event that triggered this request
 	 */
-	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp) {
+	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp, Event event) {
```
Comment (addb) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:161]: nit: Event parentEvent

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 2]
```diff
@@ -166,15 +167,16 @@ private void dispatchTrackRequest(final Map<String, Object> data, final long tim
 		eventData.put("xdm", xdmData);
 		eventData.put("data", data);
 
-		final Event event = new Event.Builder(
+		final Event xdmEvent = new Event.Builder(
```
Comment (addb) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:170]: nit: revert to event

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ void handleTrackRequest(@NonNull final Event event) {
 			return;
 		}
 
-		dispatchTrackRequest(eventData, event.getTimestamp());
+		dispatchTrackRequest(eventData, event.getTimestamp(), event);
```
Comment (addb) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:84]: nit: `dispatchTrackRequestExeperienceEvent`

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 4]
```diff
@@ -149,15 +149,16 @@ void handleRulesEngineResponse(@NonNull final Event event) {
...
 	}
 
 	/**
 	 * Helper to create and dispatch an experience event.
 	 * @param data map containing free-form data to send to Edge Network
 	 * @param timestamp timestamp of Event
+	 * @param parentEvent the parent event that triggered this request
 	 */
-	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp) {
+	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp, final Event parentEvent) {
```
Comment (emdobrin) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:161]: we can pass in just the parentEvent and extract the timestamp too: dispatchTrackRequest(final Map<String, Object> data, final Event parentEvent)

File: code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java [hunk 5]
```diff
@@ -149,18 +149,18 @@ void handleRulesEngineResponse(@NonNull final Event event) {
...
 
-		dispatchTrackRequest(detail, event.getTimestamp());
+		dispatchTrackRequest(detail, event);
 	}
 
 	/**
 	 * Helper to create and dispatch an experience event.
 	 * @param data map containing free-form data to send to Edge Network
-	 * @param timestamp timestamp of Event
+	 * @param parentEvent the parent event that triggered this request
```
Comment (emdobrin) [inline code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java:158]: can we update description here, e.g.
@param parentEvent the triggering parent event used for event chaining; its timestamp is set as xdm.timestamp

**Review by emdobrin — APPROVED** [review]

**Pull request merged**